### PR TITLE
add lifecycle and bucket logging test

### PR DESCRIPTION
**Pull request by dongbula**

### CODE REVIEW

File: src/s3.c [hunk 1]
```diff
@@ -3032,12 +3032,14 @@ static void head_object(int argc, char **argv, int optindex)
     while (*slash && (*slash != '/')) {
         slash++;
     }
-    if (!*slash || !*(slash + 1)) {
-        fprintf(stderr, "\nERROR: Invalid bucket/key name: %s\n",
-                argv[optindex]);
-        usageExit(stderr);
+    //if (!*slash || !*(slash + 1)) {
```
Comment (bji) [inline src/s3.c]: Why are you commenting this code out?

File: src/s3.c [hunk 2]
```diff
@@ -3794,49 +3797,49 @@ void set_logging(int argc, char **argv, int optindex)
                           TARGET_PREFIX_PREFIX_LEN)) {
             targetPrefix = &(param[TARGET_PREFIX_PREFIX_LEN]);
         }
-        else if (!strncmp(param, FILENAME_PREFIX, FILENAME_PREFIX_LEN)) {
-            filename = &(param[FILENAME_PREFIX_LEN]);
-        }
+        //else if (!strncmp(param, FILENAME_PREFIX, FILENAME_PREFIX_LEN)) {
```
Comment (bji) [inline src/s3.c]: Why are you commenting this code out?

File: test.diff [hunk 3]
```diff
@@ -0,0 +1,241 @@
+diff --git a/src/s3.c b/src/s3.c
```
Comment (bji) [inline test.diff]: Probably you didn't mean to add in this file.  Please remove it from your pull request.

File: test/test.sh [hunk 4]
```diff
@@ -25,7 +25,7 @@ if [ -z "$TEST_BUCKET_PREFIX" ]; then
 fi
 
 if [ -z "$S3_COMMAND" ]; then
-    S3_COMMAND=s3
+    S3_COMMAND="s3 -u"
```
Comment (bji) [inline test/test.sh]: Why are you changing this?  And without updating the comment at the beginning of the file that documents what the default value is?

File: test/test.sh [hunk 5]
```diff
@@ -78,11 +86,13 @@ echo "$S3_COMMAND delete $TEST_BUCKET"
 $S3_COMMAND delete $TEST_BUCKET
 failures=$(($failures + (($? == 0) ? 0 : 1)))
 
-# Make sure it's not there
+echo failures:$failures
```
Comment (bji) [inline test/test.sh]: What are you echoing the failures value here?

File: test/test.sh [hunk 6]
```diff
@@ -166,7 +178,7 @@ echo "$S3_COMMAND setacl $TEST_BUCKET filename=acl"
 $S3_COMMAND setacl $TEST_BUCKET filename=acl
 failures=$(($failures + (($? == 0) ? 0 : 1)))
 
-# Test to make sure that it worked
+# Test to make sure that it worked, the failures will be added when using ceph s3
```
Comment (bji) [inline test/test.sh]: libs3 does not claim compatibility with ceph s3 so not sure this comment is necessary

File: test/test.sh [hunk 7]
```diff
@@ -213,6 +225,38 @@ diff mpfile mpfile.get
 failures=$(($failures + (($? == 0) ? 0 : 1)))
 rm -f mpfile mpfile.get
 
+rm -f lifecycleconf lifecycleconf_new
```
Comment (bji) [inline test/test.sh]: As far as I can tell, these added tests are the only thing I would be likely to take out of your pull request, unless you have some justifications for the other changes.

**Comment by dongbula** [comment]
@bji I'm sorry, I have pushed it wrongly, it's just a local test prepared for using for ceph, will close it.

**Pull request closed without merging**